File: 242-valid-anagram/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        cntrS = Counter(s)
        for char in t:
            if char in cntrS:
                cntrS[char] -= 1
                if len(+cntrS) == 0:
                    return True
            else:
                return False
        return False

File: 242-valid-anagram/242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;
        int[] characters = new int[26];
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            characters[(int) c - (int) 'a']++;
        }
        for (int i = 0; i < t.length(); i++) {
            char c = t.charAt(i);
            characters[(int) c - (int) 'a']--;
        }
        return Arrays.toString(characters).equals(Arrays.toString(new int[26]));
    }
}